### PR TITLE
homework_6

### PR DESCRIPTION
**Pull request by reshetnyk**

### CODE REVIEW

File: homeworks/olexandr.reshetnyk_reshetnyk/homework_6/task_1.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        set(value) {
+            fullName = value.toLowerCase().replace(/\b\w/g, firstLetter => firstLetter.toUpperCase());
+        },
+    });
+    Object.defineProperty(obj, 'phone', {
+        get() {
+            return phone;
+        },
+        set(value) {
+            phone = value.toLowerCase().replace(/(^\+)|\D*(\d+)\D*/g, (str, plus, num) => (plus) || num);
```
Comment (markoffden) [inline homeworks/olexandr.reshetnyk_reshetnyk/homework_6/task_1.js]: hmmm, looks like easier here is to look for non-digits first and then add plus sign optionally

File: homeworks/olexandr.reshetnyk_reshetnyk/homework_6/task_2.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                // calculating winners
+                const distances = [];
+                // fill distances array off all horses
+                this.horses.forEach((horse) => {
+                    distances.push(horse.distance);
+                });
+                // calc max distance
+                const maxDistance = Math.max.apply(null, distances);
+                // find winners from Racer instances
+                const winners = this.horses.filter(horse => horse.distance === maxDistance);
```
Comment (markoffden) [inline homeworks/olexandr.reshetnyk_reshetnyk/homework_6/task_2.js]: maybe it was easier to simply `sort()` horses array by distance property?

**Comment by markoffden** [comment]
in general ok, but some parts could be done easier

**Comment by reshetnyk** [comment]
> in general ok, but some parts could be done easier

i corrected these parts

File: homeworks/olexandr.reshetnyk_reshetnyk/homework_6/task_2.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                // sorting horses by DESC their distances
+                this.horses.sort((h1, h2) => ((h1.distance < h2.distance) ? 1 : -1));
+                // print winners
+                console.log('Congratulation our horses-winners!');
+                for (let i = 0; i < this.horses.length; i++) {
+                    if (this.horses[i].distance === this.horses[0].distance) {
+                        console.log(`Winner: ${this.horses[i].name}  ${this.horses[i].distance}`);
+                    } else {
+                        break;
+                    }
```
Comment (markoffden) [inline homeworks/olexandr.reshetnyk_reshetnyk/homework_6/task_2.js]: don't need to `break` here

File: homeworks/olexandr.reshetnyk_reshetnyk/homework_6/task_2.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            });
+            // if 10 sec then finish the race
+            if (this.raceTime >= 10) {
+                this.horses.forEach((horse) => {
+                    horse.finish();
+                });
+                clearInterval(this.raceTimer);
+
+                // sorting horses by DESC their distances
+                this.horses.sort((h1, h2) => ((h1.distance < h2.distance) ? 1 : -1));
```
Comment (markoffden) [inline homeworks/olexandr.reshetnyk_reshetnyk/homework_6/task_2.js]: read once again how `sort()` works, you could avoid extra code here

**Comment by markoffden** [comment]
in general nice, but please fix minor issues as per comments

File: homeworks/olexandr.reshetnyk_reshetnyk/homework_6/task_2.js (hunk 3 above)
Comment (reshetnyk) [reply homeworks/olexandr.reshetnyk_reshetnyk/homework_6/task_2.js]: why not? its the way to escape many unnecessary iterations.

**Comment by reshetnyk** [comment]
> in general nice, but please fix minor issues as per comments

Done it.

File: homeworks/olexandr.reshetnyk_reshetnyk/homework_6/task_1.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+function addUser() {
+    let fullName;
+    let phone;
+    const obj = {};
```
Comment (markoffden) [inline homeworks/olexandr.reshetnyk_reshetnyk/homework_6/task_1.js]: better to take initial object as argument

**Comment by markoffden** [comment]
accepted, but both tasks could be optimized

**Pull request merged**